### PR TITLE
Address long-running queries and OOM contributor in PHP replacement

### PR DESCRIPTION
**Pull request by brandonpayton**
This change makes PHP replacement queries and memory usage more efficient in order to address issues we've encountered with long-running queries and OOM conditions with certain sites.

It improves the primary key SELECT queries by eliminating the use of OFFSET because OFFSET requires that the database consider all rows up to OFFSET before taking rows up to the LIMIT. This can make queries become slower as OFFSET increases. The new SELECT query relies on primary key conditions to more efficiently eliminate previous rows from consideration. This way, the database can use an index to identify rows with keys greater than those of the previous chunk and then take rows from that set up to the LIMIT.

It improves memory usage by doing updates along the way rather than storing all a column's updated values in memory until the end. At Automattic, when we limit search-replace to 4GB of memory, we sometimes exceed that limit for large sites. It's possible there are other things that contribute to high memory usage within the search-replace command, but as a first step, we can reduce memory requirements by no longer keeping all updated column values in memory simultaneously.

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@brandonpayton Has this been running in production for some time? Also, are there any particular test cases you think we should add to cover the change?

**Comment by brandonpayton** [comment]
@danielbachhuber, this is not running in production yet but should be next week. I'll withdraw this PR for now and plan to resubmit later after it has been in production for a while. I figured the current test cases would cover this code as the resulting replacements should not change, but it would be a good idea for me to review the tests and confirm that. I can do that before resubmitting the PR. Thanks!

**Comment by danielbachhuber** [comment]
> I'll withdraw this PR for now and plan to resubmit later after it has been in production for a while.

@brandonpayton Sounds good, thanks!

> I figured the current test cases would cover this code as the resulting replacements should not change, but it would be a good idea for me to review the tests and confirm that.

I think the functional existing tests are _pretty_ good but may or may not be comprehensive enough to cover this. It could be fine or there could be some minor idiosyncrasy that causes problems.

File: src/Search_Replace_Command.php [hunk 1]
```diff
@@ -595,15 +597,22 @@ static function ( $key ) {
...
+					$wpdb->update( $table, [ $col => $value ], $update_where );
 				}
 			}
 
-			$offset += $limit;
-		}
-
-		foreach ( $updates as $update ) {
-			$wpdb->update( ...$update );
+			// Because we are ordering by primary keys from least to greatest,
```
Comment (aidvu) [inline src/Search_Replace_Command.php:604]: Love it, 2 wins. Runs faster than limit/offset and solves the memory problems. Thanks for working on it. 🙇🏻

**Comment by brandonpayton** [comment]
Actually, I do not have permission to reopen. :)  @danielbachhuber, are you up for reopening this PR?

**Comment by danielbachhuber** [comment]
@brandonpayton It seems like the branch disappeared for me too:

image

Want to create a new PR?

**Comment by brandonpayton** [comment]
> I'll withdraw this PR for now and plan to resubmit later after it has been in production for a while.

It was the originally state plan anyway, but I figured "why not just reopen?"  I guess this is why :)

Thanks for looking, @danielbachhuber!

**Pull request closed without merging**